src/components/partner/partner.service.ts

```diff
@@ -136,14 +145,49 @@ export class PartnerService {
       await this.repo.updateRelation(
         'languageOfWiderCommunication',
         'Language',
-        object.id,
+        partner.id,
         languageOfWiderCommunicationId,
       );
     }
 
+    if (countries?.length && countries.length > 0) {
+      await this.validateCountryIds(
+        countries,
+        'Countries',
+        'countryId',
+        'Country not found',
+      );
+
+      await this.repo.updateRelationList({
+        id: partner.id,
+        relation: 'countries',
+        newList: countries,
+      });
+    }
+
     return await this.readOne(input.id, session);
   }
 
+  protected async validateCountryIds(
+    ids: Many<ID> | null | undefined,
+    label: string,
+    resourceField: string,
+    errMsg: string,
+  ): Promise<void> {
+    await Promise.all(
+      many(ids ?? []).map(async (id, index) => {
+        const country = await this.repo.getBaseNode(id, label);
+        if (country) {
+          return;
+        }
+        throw new NotFoundException(
+          errMsg,
+          `partner.${resourceField}${Array.isArray(ids) ? `[${index}]` : ''}`,
+        );
+      }),
+    );
+  }
+
```

Comment:
For this whole section you'll be using an abstraction now that handles most of this; you're code will be much simpler.  You can model it after [the example here](https://github.com/SeedCompany/cord-api-v3/blob/develop/src/components/partner/partner.service.ts#L146C2-L158C2).  Note that you'll have to rebase on `develop` to get Carson's abstraction code before you can make these changes.